### ISSUE TITLE
interfaces enabled="false" not working in hazelcast 3.8.6

### ISSUE THREAD

**Issue by prasanthnath** [hazelcast/hazelcast#11442]
Hi,

We are using Hazelcast 3.7.4 so far and we have the following hazelcast configuration:
```
<?xml version="1.0" encoding="UTF-8"?>
<hazelcast xsi:schemaLocation="http://www.hazelcast.com/schema/config hazelcast-config-3.7.xsd"
           xmlns="http://www.hazelcast.com/schema/config"
           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
   <group>
      <name>taxilla_dev_prasanth</name>
      <password>dev@123</password>
   </group>
   <management-center enabled="false">http://localhost:8080/mancenter</management-center>
   <properties>
      <property name="hazelcast.logging.type">slf4j</property>
      <property name="hazelcast.socket.bind.any">false</property>
   </properties>
   <network>
      <join>
         <multicast enabled="false"/>
         <tcp-ip enabled="true">
            <member-list>
               <member>192.168.3.62</member>
            </member-list>
         </tcp-ip>
      </join>
      <interfaces enabled="false">
         <interface>192.168.*.*</interface>
      </interfaces>
   </network>
</hazelcast>
```
So far this is working fine. I tried upgrading to hazelcast 3.8.6 in one of the servers. In other servers, I have the older version running. Now when I tried to start up application with hazelcast 3.8.6, it is giving me the following exception:
 
```
68 - [192.168.3.62]:5701 [taxilla_dev_prasanth] [3.8.6] Failed to process packet: Packet{partitionId=-1, conn=Connection[id=1, /192.168.3.62:5701->/192.168.1.102:53214, endpoint=[192.168.1.102]:5702, alive=true, type=MEMBER], rawFlags=10001, isUrgent=true, packetType=OPERATION, typeSpecificFlags=[isResponse=false, isOpControl=false]} on hz._hzInstance_1_taxilla_dev_prasanth.priority-generic-operation.thread-0
com.hazelcast.nio.serialization.HazelcastSerializationException: Problem while reading DataSerializable, namespace: 0, id: 0, class: 'com.hazelcast.internal.cluster.impl.operations.JoinCheckOperation', exception: com.hazelcast.internal.cluster.impl.operations.JoinCheckOperation
        at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.rethrowReadException(DataSerializableSerializer.java:177)
        at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.readInternal(DataSerializableSerializer.java:158)
        at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.read(DataSerializableSerializer.java:103)
        at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.read(DataSerializableSerializer.java:49)
        at com.hazelcast.internal.serialization.impl.StreamSerializerAdapter.read(StreamSerializerAdapter.java:48)
        at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:184)
        at com.hazelcast.spi.impl.NodeEngineImpl.toObject(NodeEngineImpl.java:350)
        at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:386)
        at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:115)
        at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:100)
Caused by: java.lang.ClassNotFoundException: com.hazelcast.internal.cluster.impl.operations.JoinCheckOperation
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at org.mortbay.jetty.webapp.WebAppClassLoader.loadClass(WebAppClassLoader.java:401)
        at org.mortbay.jetty.webapp.WebAppClassLoader.loadClass(WebAppClassLoader.java:363)
        at com.hazelcast.nio.ClassLoaderUtil.tryLoadClass(ClassLoaderUtil.java:149)
        at com.hazelcast.nio.ClassLoaderUtil.loadClass(ClassLoaderUtil.java:123)
        at com.hazelcast.nio.ClassLoaderUtil.newInstance(ClassLoaderUtil.java:77)
        at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.readInternal(DataSerializableSerializer.java:147)
        ... 8 common frames omitted
```
From the above log, I can see the server is trying to cluster with other servers which has got older implementations. I am not able to understand why this is happening as I have disabled it by specifying `<interfaces enabled="false">`

When I switch back to older version of hazel cast, I don’t see any issues. Is there some configuration that I am doing incorrectly or is this an issue with the implementation? Please confirm!

**Comment by prasanthnath** [comment]
Do I have any updates on this?

**Comment by jerrinot** [comment]
@prasanthnath: rolling upgrade from 3.6 to 3.8 is not supported. 
You can use a different client version, but all cluster members have to be at the same minor versions.

**Comment by prasanthnath** [comment]
@jerrinot We have upgraded from 3.7.4 to 3.8.6 and not from 3.6
Also would you mind explaining what you mean by different client version? I could not understand much from the comment. Can you provide more details?

**Comment by mmedenjak** [comment]
@prasanthnath rolling upgrade from 3.7.x to 3.8.x is not supported as well. Rolling upgrades are available from 3.8 onwards. 
I think what he referred to was that in the stable state, the clients may be of different minor versions than the member versions (e.g. you can have 3.8 members and 3.7 clients) but cluster members need to be at the same minor version.
The only time members don't need to be at the same minor version is when you are in the process of rolling upgrades. Then the members should differ by at most 1 minor version (e.g. you can have a mixed cluster of 3.8 and 3.9 members but not 3.8 and 3.10).
